File: Backend/src/data/entities/Message.ts
import { IsNotEmpty, IsString } from "class-validator";
import { ApiProperty } from "@nestjs/swagger";
import { uuid } from "../../utils/util";
import { MessageCreateDto } from "../../api/messages/models/requests/MessageCreate.dto";

export class Message {
@ApiProperty()
id: string;
@ApiProperty()
body: string;
@ApiProperty()
teamId: string;
@ApiProperty()
senderName: string;

static fromDto(dto: MessageCreateDto, teamId: string, senderName: string): Message {
		return {
			id: uuid(),
			body: dto.body,
			teamId: teamId,
            senderName: senderName,
		};
	}
}

File: frontend/lib/Types/message.dart
class Message {
  String id;
  String body;
  String teamId;
  String senderName;

  Message(this.id, this.body, this.teamId, this.senderName);

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      json['id'],
      json['body'],
      json['teamId'],
      json['senderName'],
    );
  }
}
